Adaptors/Amqp/src/PushQueueStorage.cs

```diff
@@ -96,37 +94,43 @@ public class PushQueueStorage : QueueStorage, IPushQueueStorage
-
-                                         await sender.SendAsync(new Message(Encoding.UTF8.GetBytes(msgData.TaskId))
-                                                                {
-                                                                  Header = new Header
-                                                                           {
-                                                                             Priority = (byte)internalPriority,
-                                                                           },
-                                                                  Properties = new Properties
-                                                                               {
-                                                                                 MessageId = Guid.NewGuid()
-                                                                                                 .ToString(),
-                                                                               },
-                                                                })
-                                                     .ConfigureAwait(false);
+                                         var address = $"{partitionId}###q{whichQueue}";
+
+                                         var pool = senders_.GetOrAdd(address,
+                                                                      s => new ObjectPool<SenderLink>(parallelismLimit_,
+                                                                                                      async token => new SenderLink(new Session(await ConnectionAmqp
+                                                                                                                                                      .GetConnectionAsync(token)
+                                                                                                                                                      .ConfigureAwait(false)),
+                                                                                                                                    Guid.NewGuid()
+                                                                                                                                        .ToString(),
+                                                                                                                                    s),
+                                                                                                      (link,
+                                                                                                       _) => new ValueTask<bool>(!link.IsClosed &&
+                                                                                                                                 !link.Session.IsClosed)));
+
+                                         await pool.WithInstanceAsync(sender => sender.SendAsync(new Message(Encoding.UTF8.GetBytes(msgData.TaskId))
```

Comment:
You did not pass the cancellation token to SendAsync.

Reply:
It does not take cancellation tokens